### PR TITLE
dev/imageless-pipeline

### PR DESCRIPTION
**Pull request by borovik135**
a version of the pipeline that works with just the point cloud, no input imagery

### CODE REVIEW

File: tools/run_danesfield.py [hunk 1]
```diff
@@ -473,7 +473,7 @@ def main(args):
                  cmd_args)
 
     texture_mapping_outdir = os.path.join(working_dir, 'texture-mapping')
-    occlusion_mesh = "xxxx.obj"
+    occlusion_mesh = os.path.join(roof_geon_extraction_outdir, 'occlusion_mesh.obj')
```
Comment (dmjoy) [inline tools/run_danesfield.py:476]: I don't recall exactly the purpose / origin "xxxx.obj" file, but I don't know that this was a typo.  Are these two files by chance the same, one just has a much more descriptive name?

**Review by dmjoy — COMMENTED** [review]
At a high level the added scripts look good to me, but I don't see where they're being called in the `run_danesfield_imageless.py` script.

Also, given that there seems to be a significant amount of overlap between `run_danesfield.py` and `run_danesfield_imageless.py` make just adding an argument (or configuration option) onto `run_danesfield.py` for running "imageless" mode makes the most sense?

File: tools/run_danesfield.py (hunk 1 above)
Comment (borovik135) [reply tools/run_danesfield.py:476]: The legacy code would output xxxx.obj to the working directory, and with such name it was hard to tell what the file was and its purpose. The modification ensures that all intermediate and final output gets where it belongs, i.e. respective output directory.

**Review by borovik135 — COMMENTED** [review]
> At a high level the added scripts look good to me, but I don't see where they're being called in the `run_danesfield_imageless.py` script.

Some of the added scripts are utilities that may be optionally run before or after the actual pipeline. The cloud clean up should probably be run just once for a particular cloud. Cloud GUI visualization is also optional, could not be invoked in the headless mode, and thus should not be part of the pipeline.

> Also, given that there seems to be a significant amount of overlap between `run_danesfield.py` and `run_danesfield_imageless.py` make just adding an argument (or configuration option) onto `run_danesfield.py` for running "imageless" mode makes the most sense?

I had similar after-thoughts, but during our image-less pipeline development we wanted to deprive the script of any source image input to identify and remove the blocks that depend on it, hence the separate script. Going forward, we could merge the two and have an option of running image-less, as suggested.

File: tools/run_danesfield.py (hunk 1 above)
Comment (dmjoy) [reply tools/run_danesfield.py:476]: Ah I see, this is setting the output path for the occlusion mask.  Totally on board with this change then, thanks.

**Comment by dmjoy** [comment]
I see, though in theory in sounds like the cloud cleanup step could be an optional first step of the imageless pipeline (assuming it doesn't modify the pointcloud in place).

Regarding combining the two scripts, If there's a time crunch I think we could probably push that off into a follow-up PR, but I do think it will make maintenance easier in the long run.

**Comment by borovik135** [comment]
Cloud clean-up should then be an option for both image-less/ful pipelines; or for a single pipeline that can optionally run image-less. Those improvements may be left for another PR that is likely to happen after the upcoming demo, I think.

**Comment by dmjoy** [comment]
Gotcha.  I'll go ahead and merge then and we can make those tweaks in a follow-up PR.

**Pull request merged**